app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt

```diff
@@ -0,0 +1,102 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+   License, v. 2.0. If a copy of the MPL was not distributed with this
+   file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.history
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.navigation.Navigation
+import kotlinx.android.synthetic.main.fragment_history.view.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.requireComponents
+import org.mozilla.fenix.mvi.ActionBusFactory
+import org.mozilla.fenix.mvi.getSafeManagedObservable
+import kotlin.coroutines.CoroutineContext
+
+class HistoryFragment : Fragment(), CoroutineScope {
+
+    lateinit var job: Job
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Main + job
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_history, container, false)
+        HistoryComponent(view.history_layout, ActionBusFactory.get(this), initialState = HistoryState(emptyList()))
+
+        return view
+    }
+
+    @SuppressLint("CheckResult")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        job = Job()
+
+        setHasOptionsMenu(true)
+        (activity as AppCompatActivity).supportActionBar?.show()
+
+        getSafeManagedObservable<HistoryAction>()
+            .subscribe {
+                if (it is HistoryAction.Select) {
+                    Navigation.findNavController(requireActivity(), R.id.container)
+                        .popBackStack(R.id.browserFragment, false)
+
+                    requireComponents.useCases.sessionUseCases.loadUrl.invoke(it.item.url)
+                }
+            }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        job.cancel()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.library_menu, menu)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val eventEmitter = ActionBusFactory.get(this)
+
+        launch {
```

Comment:
I'd specify which coroutineDispatcher you want here. If you're doing background work, it would make sense to use the IO dispatcher. CommonPool is usually wrong on Android. Your coroutineContext was set to default to the Main dispatcher.

FYI, unlike async, exceptions inside of launch don't cause the parent process to error unless you join() on the Job. This is just a quirk to be aware of.